### PR TITLE
[DMS-7] Initialize application logging

### PR DESCRIPTION
**Pull request by CSR2017**

### CODE REVIEW

File: src/services/EdFi.DataManagementService.Api/Program.cs [hunk 1]
```diff
@@ -3,21 +3,29 @@
...
 app.MapGet("/ping", () => Results.Ok(DateTime.Now));
 
-var logAppSettingsService = app.Services.GetRequiredService<LogAppSettingsService>();
-logAppSettingsService.LogToConsole();
+app.MapGet("/LogAppSettings", (IOptions<AppSettings> options, ILogger<Program> logger) =>
+{
+    logger.LogDebug($"Log while debugging");
+    logger.LogInformation(message: JsonSerializer.Serialize(options.Value));
+    return Results.Ok("Successfully logged Application Settings");
+});
```
Comment (stephenfuqua) [inline src/services/EdFi.DataManagementService.Api/Program.cs:28]: Is there a plan to take this out again in the future, before this code goes to production?

Comment (CSR2017) [reply src/services/EdFi.DataManagementService.Api/Program.cs:29]: Yes @stephenfuqua. I am keeping that endpoint just to validate the  IOptions and  ILogger !

File: src/services/EdFi.DataManagementService.Api/Infrastructure/LoggingMiddleware.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+namespace EdFi.DataManagementService.Api.Infrastructure;
+
+public class LoggingMiddleware
+{
+    private readonly RequestDelegate _next;
+
+    public LoggingMiddleware(RequestDelegate next)
+    {
+        _next = next ?? throw new ArgumentNullException(nameof(next));
```
[the remark this answers is not in the document]
Comment (CSR2017) [reply src/services/EdFi.DataManagementService.Api/Infrastructure/LoggingMiddleware.cs:17]: Looks like it is uncommon to have "RequestDelegate" to be null.

[the remark this answers is not in the document]
Comment (stephenfuqua) [reply src/services/EdFi.DataManagementService.Api/Infrastructure/LoggingMiddleware.cs:17]: In fact, it is impossible for it to be `null`. Try setting `next = null` and you'll find that it won't compile. Therefore the null check is unnecessary.

File: src/services/EdFi.DataManagementService.Api/Infrastructure/LoggingMiddleware.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    public async Task Invoke(HttpContext context, ILogger<LoggingMiddleware> logger)
+    {
+        try
+        {
+            context = context ?? throw new ArgumentNullException(nameof(context));
```
[the remark this answers is not in the document]
Comment (CSR2017) [reply src/services/EdFi.DataManagementService.Api/Infrastructure/LoggingMiddleware.cs]: Removing the null check. HttpContext generally not null.

**Comment by simpat-adam** [comment]
@CSR2017 One of the acceptance criteria says "The interface should include convenience properties IsDebugEnabled and IsInfoEnabled." I didn't see such a thing here.

**Comment by CSR2017** [comment]
> @CSR2017 One of the acceptance criteria says "The interface should include convenience properties IsDebugEnabled and IsInfoEnabled." I didn't see such a thing here.

@simpat-adam I discussed with @bradbanister. The purpose is to customize the log levels according to the environment. Since, we are having different app-settings now, the development log level is always in "DEbug". The production one is customizable.

**Comment by bradbanister** [comment]
> > @CSR2017 One of the acceptance criteria says "The interface should include convenience properties IsDebugEnabled and IsInfoEnabled." I didn't see such a thing here.
> 
> @simpat-adam I discussed with @bradbanister. The purpose is to customize the log levels according to the environment. Since, we are having different app-settings now, the development log level is always in "DEbug". The production one is customizable.

Maybe I misunderstood. We do want to have those properties exposed.

File: src/services/EdFi.DataManagementService.Api/appsettings.json [hunk 4]
```diff
@@ -1,21 +1,37 @@
...
-    },
-    "Logging": {
-        "LogLevel": {
-            "Default": "Information",
-            "Microsoft.AspNetCore": "Warning"
-        }
-    },
-    "AllowedHosts": "*",
-    "RateLimit": {
+  "AppSettings": {
```
[the remark this answers is not in the document]
Comment (stephenfuqua) [reply src/services/EdFi.DataManagementService.Api/appsettings.json]: 2 spaces is my personal preference for JSON and YAML.

https://github.com/Ed-Fi-Alliance-OSS/Data-Management-Service/blob/main/.editorconfig is setup for `indent_size: 4` except on YAML. The YAML section should also list JSON. I just missed  that when I copied that .editorconfig in from another location.

[the remark this answers is not in the document]
Comment (CSR2017) [reply src/services/EdFi.DataManagementService.Api/appsettings.json]: Updated the common editor config to have indent size =2 for .json files

**Pull request merged**